growthbook/growthbook
# Improve array text inputs on GB

### Features and Changes

Replaces text fields in GrowthBook that require users to manually enter lists of comma separated values with Tokenized input fields using the existing `StringArrayField` component. Replaces the text fields for the following components:

- ConditionInput
- SavedGroupForm
- InviteModal

For `ConditionInput` and `SavedGroupForm` we give users the option to switch back to raw text input to support cases where users might be copying & pasting a long list of CSV's. We'll have some separate work later to allow users to upload CSVs for these fields rather than copy & paste.

This PR also resolves a small pre-existing bug within the saved group controller (see my comment in the code) as well as some minor code quality improvements. 

Closes #1427 

### Testing

1. Ensure that fields work as they did before aside from UX improvements
2. Make sure values are persisting correctly after completing a save
3. Test alternating between the token field and the raw text field and make sure values look the same on both fields

### Screenshots

New Experience
<img width="776" alt="Screenshot 2023-07-27 at 9 44 49 AM" src="https://github.com/growthbook/growthbook/assets/14947905/833519e6-441e-48af-a905-c0cf08ad4d30">
<img width="779" alt="Screenshot 2023-07-27 at 9 45 01 AM" src="https://github.com/growthbook/growthbook/assets/14947905/db0ee86e-6e52-4bdd-8577-97cf14e2f7eb">


Previous Experience
<img width="780" alt="Screenshot 2023-07-20 at 4 17 14 PM" src="https://github.com/growthbook/growthbook/assets/14947905/c29c2ac3-6f71-4087-b43a-7e96c93fc158">
